File: 2015-determine-whether-matrix-can-be-obtained-by-rotation/determine-whether-matrix-can-be-obtained-by-rotation.py
class Solution:
    def findRotation(self, mat: List[List[int]], target: List[List[int]]) -> bool:

        def rotate(mat):
            n = len(mat)

            com = [[0]*n for _ in range(n)]

            for i in range(n):
                for j in range(n):
                    com[j][n-1-i] = mat[i][j]
            
            return com
        
        def compare(mat,target):
            n=len(mat)
            for i in range(n):
                for j in range(n):
                    if mat[i][j] != target[i][j]:
                        return False
            
            return True
            
        
        for _ in range(4):
            ans = compare(mat,target)
            if ans :
                return True
            mat = rotate(mat)
        
        return False
            
                

        

File: 2015-determine-whether-matrix-can-be-obtained-by-rotation/determine-whether-matrix-can-be-obtained-by-rotation.java
class Solution {
    public boolean compare(int[][] mat,int[][] target){

        int n = mat.length;
        for(int i=0;i<n;i++){
            for(int j=0;j<n;j++){
                if(mat[i][j] != target[i][j]){
                    return false;
                }
            }
        }
        return true;

    }

    public int[][] routate(int[][] mat){
        int n = mat.length;
        int[][] com = new int[n][n];
        for(int i=0;i<n;i++){
            for(int j=0;j<n;j++){
                com[j][n-1-i] = mat[i][j];
            }
        }
        return com;
    }
    public boolean findRotation(int[][] mat, int[][] target) {

        for(int i=0 ;i<4;i++){
            boolean ans = compare(mat,target);
            if (ans == true){
                return true;
            }
            mat = routate(mat);
        }

        return false;   
    }
}